File: backend/app/schemas/cliente.py
"""Schemas para Cliente."""

from datetime import datetime
from typing import Optional
from pydantic import BaseModel, EmailStr, Field


class ClienteBase(BaseModel):
    """Base schema para Cliente."""
    razon_social: str = Field(..., min_length=1, max_length=255)
    tipo_documento: str = Field(..., pattern=r"^(CUIT|CUIL|DNI|LE|LC|Pasaporte|CI)$")
    numero_documento: str = Field(..., min_length=1, max_length=20)
    condicion_iva: str = Field(..., pattern=r"^(RI|Monotributo|CF|Exento)$")
    domicilio: Optional[str] = Field(None, max_length=255)
    localidad: Optional[str] = Field(None, max_length=100)
    provincia: Optional[str] = Field(None, max_length=100)
    codigo_postal: Optional[str] = Field(None, max_length=10)
    email: Optional[EmailStr] = None
    telefono: Optional[str] = Field(None, max_length=50)
    notas: Optional[str] = Field(None, max_length=500)


class ClienteCreate(ClienteBase):
    """Schema para crear Cliente."""
    pass


class ClienteUpdate(BaseModel):
    """Schema para actualizar Cliente."""
    razon_social: Optional[str] = Field(None, min_length=1, max_length=255)
    tipo_documento: Optional[str] = Field(None, pattern=r"^(CUIT|CUIL|DNI|LE|LC|Pasaporte|CI)$")
    numero_documento: Optional[str] = Field(None, min_length=1, max_length=20)
    condicion_iva: Optional[str] = Field(None, pattern=r"^(RI|Monotributo|CF|Exento)$")
    domicilio: Optional[str] = Field(None, max_length=255)
    localidad: Optional[str] = Field(None, max_length=100)
    provincia: Optional[str] = Field(None, max_length=100)
    codigo_postal: Optional[str] = Field(None, max_length=10)
    email: Optional[EmailStr] = None
    telefono: Optional[str] = Field(None, max_length=50)
    notas: Optional[str] = Field(None, max_length=500)
    activo: Optional[bool] = None


class ClienteResponse(ClienteBase):
    """Schema de respuesta de Cliente."""
    id: int
    empresa_id: int
    activo: bool
    created_at: datetime
    updated_at: datetime
    
    class Config:
        from_attributes = True


class ClienteList(BaseModel):
    """Schema para listado paginado de Clientes."""
    items: list[ClienteResponse]
    total: int
    page: int
    per_page: int
    pages: int

File: frontend/src/types/cliente.ts
export interface Cliente {
  id: number
  empresa_id: number
  razon_social: string
  tipo_documento: 'CUIT' | 'CUIL' | 'DNI' | 'LE' | 'LC' | 'Pasaporte' | 'CI'
  numero_documento: string
  condicion_iva: 'RI' | 'Monotributo' | 'CF' | 'Exento'
  domicilio: string | null
  localidad: string | null
  provincia: string | null
  codigo_postal: string | null
  email: string | null
  telefono: string | null
  notas: string | null
  activo: boolean
  created_at: string
  updated_at: string
}

export interface ClienteCreate {
  razon_social: string
  tipo_documento: 'CUIT' | 'CUIL' | 'DNI' | 'LE' | 'LC' | 'Pasaporte' | 'CI'
  numero_documento: string
  condicion_iva: 'RI' | 'Monotributo' | 'CF' | 'Exento'
  domicilio?: string
  localidad?: string
  provincia?: string
  codigo_postal?: string
  email?: string
  telefono?: string
  notas?: string
}

export interface ClienteUpdate {
  razon_social?: string
  tipo_documento?: 'CUIT' | 'CUIL' | 'DNI' | 'LE' | 'LC' | 'Pasaporte' | 'CI'
  numero_documento?: string
  condicion_iva?: 'RI' | 'Monotributo' | 'CF' | 'Exento'
  domicilio?: string
  localidad?: string
  provincia?: string
  codigo_postal?: string
  email?: string
  telefono?: string
  notas?: string
  activo?: boolean
}

export interface ClienteListParams {
  page?: number
  per_page?: number
  search?: string
  activo?: boolean
}
